xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp

```diff
@@ -149,41 +149,44 @@ bool CDVDDemuxClient::ParsePacket(DemuxPacket* pkt)
     stream->m_context->time_base.den = DVD_TIME_BASE;
   }
 
-  if (stream->m_parser_split && stream->m_parser->parser->split)
+  if (stream->m_parser_split && stream->m_parser && stream->m_parser->parser)
   {
-    int len = stream->m_parser->parser->split(stream->m_context, pkt->pData, pkt->iSize);
-    if (len > 0 && len < FF_MAX_EXTRADATA_SIZE)
+    uint8_t* oldExtraData = st->ExtraData;
+
+    AVPacket* avpkt = av_packet_alloc();
+    if (!avpkt)
+    {
+      CLog::LogF(LOGERROR, "CDVDDemuxClient::{} - av_packet_alloc failed: {}", __FUNCTION__,
```

Comment:
```suggestion
      CLog::LogF(LOGERROR, "av_packet_alloc failed: {}", strerror(errno));
```

The` __FUNCTION__` parameter is not needed since `LogF` is used